clients/src/test/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumerTest.java

```diff
@@ -1748,8 +1767,7 @@ public void testMultipleBackgroundErrors() {
         final KafkaException expectedException2 = new KafkaException("Spam, Spam, Spam");
         final ErrorEvent errorEvent2 = new ErrorEvent(expectedException2);
         backgroundEventQueue.add(errorEvent2);
-        consumer.assign(singletonList(new TopicPartition("topic", 0)));
-        final KafkaException exception = assertThrows(KafkaException.class, () -> consumer.poll(Duration.ZERO));
+        final KafkaException exception = assertThrows(KafkaException.class, () -> consumer.assign(singletonList(new TopicPartition("topic", 0))));
```

Comment:
oh I hadn't realized this undesired side effect of processing background events on assign, but of course, I missed that. I would say we don't want this change in behaviour. 

We only need to ensure that the assign does not return until the background event completes, so not really need to process background events, we just need `applicationEventHandler.addAndGet(assignmentChangeEvent)` (and I would ensure we create the event using the `defaultApiTimeout` for the deadline. Makes sense? the we would keep the same behaviour we had on this test (assign does not throw any background event, poll does).

Just for the record, on the unsubscribe we do need to process background events because of the rebalanceCallbacks that run in the app thread, but that's not the case here.

Reply:
Good catch. Updated it. Thank you.